hotsos/core/ycheck/engine/properties/requires/types/snap.py

```diff
@@ -22,19 +22,45 @@ def installed_revisions(self):
         _revisions = []
         for p in self.installed:
             _revisions.append(self.packaging_helper.get_revision(p))
+
         return _revisions
 
-    def package_revision_within_ranges(self, pkg, ranges):
+    @cached_property
+    def installed_channels(self):
+        _channels = []
+        for p in self.installed:
+            _channels.append(self.packaging_helper.get_channel(p))
+
+        return _channels
+
+    def package_info_matches(self, pkg, pkg_info):
+        """
+        If snap package has versions and/or channels we check them here.
+        """
         result = False
-        revision = int(self.packaging_helper.get_revision(pkg))
-        # compare revisions for the time being
-        for item in sorted(ranges, key=lambda i: i['max'],
-                           reverse=True):
-            r_max = int(item['max'])
-            r_min = int(item['min'])
-            if r_min <= revision <= r_max:
-                result = True
-        return result
+        if any(i.get('max') is None for i in pkg_info):
+            log.warning("one or more range is missing a 'max' value so cannot "
+                        "check versions")
+        else:
+            revision = int(self.packaging_helper.get_revision(pkg))
+            # compare revisions for the time being
+            for item in sorted(pkg_info, key=lambda i: i['max'],
+                               reverse=True):
+                r_max = int(item['max'])
+                r_min = int(item['min'])
+                if r_min <= revision <= r_max:
+                    result = True
+
+            return result
```

Comment:
Should this be removed so the channel check is run?

Reply:
yes